### ISSUE TITLE
[iOS] On closing modal by pressing outside - triggered onDonePress function

### ISSUE THREAD

**Issue by St1ma** [lawnstarter/react-native-picker-select#236]
**Describe the bug**<br />
The onDonePress function should not be triggered on closing picker modal on iOS

**To Reproduce**<br />
1. Open Picker modal on iOS device
2. Change selected value
3. Press outside the modal area to close/dismiss changes
4. onDonePress function was triggered 

**Expected behavior**<br />
onDonePress function should not be called
onCancel function triggered (optional)

**Suggested solution**<br />
Please, remove onDismiss prop here - https://github.com/lawnstarter/react-native-picker-select/blob/master/src/index.js#L425
And add calling of onDonePress function here - https://github.com/lawnstarter/react-native-picker-select/blob/master/src/index.js#L345

**Additional details**<br />
-   Device: any iPhone
-   OS: iOS12+
-   react-native-picker-select version: 6.3.3
-   react-native version: 0.61.1
-   react version: 16.9.0

**Reproduction and/or code sample**<br />
```
<PickerSelect
        value={this.state.selected}
        onValueChange={(selected) => {
          this.setState({ selected });
        }}
        items={items}
        onDonePress={() => {
          console.log('done pressed') // message will be shown even on modal cancel request
          onValueChange(this.state.selected);
        }}
      >
        // some label
      </PickerSelect>
```

**Comment by lfkwtz** [comment]
seems reasonable - looks like there may have been a regression on this in #177

**Comment by St1ma** [comment]
@lfkwtz the onDismiss prop make the same done triggering as was before on pressing area outside picker
https://github.com/lawnstarter/react-native-picker-select/pull/177/files#diff-1fdf421c05c1140f6d71444ea2b27638R398
I'm not sure it should be here

**Comment by St1ma** [comment]
@lfkwtz Any updates?

**Comment by lfkwtz** [comment]
will fix this when I have a moment - feel free to submit a PR

**Comment by ruizhangcool** [comment]
This bug is fixed? i am getting same issue when i will click "Done" button. please update it soon. :)

**Comment by lfkwtz** [comment]
@St1ma @ruizhangcool please test this beta out to see if the issue is resolved and let me know: https://www.npmjs.com/package/react-native-picker-select/v/7.0.0-beta.0

**Comment by St1ma** [comment]
@lfkwtz Works perfect
Many thanks

**Comment by lfkwtz** [comment]
fixed in 6.3.4
